api/proto/metrics/metrics.proto

```diff
@@ -20,8 +20,40 @@ enum ZmetricTypes {
 message memoryMetric {
   uint32 usedMem = 2;            //in MBytes
   uint32 availMem = 3;           //in MBytes
-  double usedPercentage = 4;
-  double availPercentage = 5;
+  double usedPercentage = 4; // Obsolete - No Longer Used.
+  double availPercentage = 5; // Obsolete - No Longer Used.
+}
+
+// App Memory Metric
+message AppMemoryMetric {
+    // totalMB - Memory allocated to app instance by EVE. In Steady State,
+    //  this is expected to be same as in App instance config. Eve should
+    //  report what it actually allocated / reserved instead of what is configured.
+    uint32 totalMB = 1;
+
+    // usedMB - Memory Used by app from totalMB Memory
+    //  usedMB <= totalMB in normal cases.
+    //  But usedMB can exceed totalMB - this could be a warning to the user.
+    uint32 usedMB = 2;
+}
+
+message DeviceMemoryMetric {
+  // totalMB - Total Device memory as seen by EVE.
+  uint32 totalMB = 1;
+
+  // allocated_appsMB - Total memory Allocated to AppInstances
+  uint32 allocated_appsMB = 2;
```

Comment:
I though you were going to drop this and have the controller decide whether it will include inactive app instances or not when it calculates some total.

If you are going to keep it in the API you must specify whether or not inactive app instances are included and potentially also when an app instance in the process of being deployed (or deleted) has its count added (or deleted) from this total. Do you add immediately even though it might take a long time to download the image? etc.
Thus I think it is easier and avoids having more policy in EVE if you avoid introducing this field in the API.

Reply:
From what I remember - we decided to include this. This is all the memory on EVE that is currently allocated to Apps - that is not available for other purposes - in that moment. Policy doesn't matter here - right?
I want to keep this separate from policy - and a bit dependent on Eve implementation.

For controller - this just means - at the moment metrics message is received - here is the memory allocated for Apps. If one of the apps is in the middle of deletion - in the next metrics message, this memory will go down.

The policy part is different:

1) Currently, as I understand, De-Activated apps are also getting allocated memory - so they will be included in the calculation.
2) Tomorrow, if we change that policy - if De-activated apps are not included, that will not be included. 

But as far as this number goes - this is irrespective of the policy. THis is - at that moment - the amount of memory allocated for apps.

IIRC - this is what we discussed on Friday.

Reply:
Made Changes as discussed.